src/test/java/net/minecraftforge/debug/world/DataPackRegistriesTest.java

```diff
@@ -0,0 +1,309 @@
+        public static void onRightClickBlock(RightClickBlock event)
+        {
+            if (ENABLED)
+            {
+                if (event.getPlayer() instanceof LocalPlayer player)
+                {
+                    RegistryAccess registries = player.connection.registryAccess();
+                    DataPackRegistriesTest.printSyncedRegistry(registries);
+                }
+            }
+        }
+    }
+
+    public static class TestA extends ForgeRegistryEntry<TestA>
+    {
+        public static final ResourceKey<Registry<TestA>> REGISTRY = ResourceKey.createRegistryKey(new ResourceLocation(MODID, MODID + "/worldgen/test_a"));
+
+        public static final Codec<TestA> DIRECT_CODEC = RecordCodecBuilder.create(instance -> instance.group(
+                Codec.STRING.fieldOf("string_value").forGetter(TestA::getStringValue),
+                Codec.INT.fieldOf("int_value").forGetter(TestA::getIntValue),
+                Biome.CODEC.fieldOf("biome").forGetter(TestA::getBiome)
+            ).apply(instance, TestA::new));
+
+        public static final RegistryFileCodec<TestA> CODEC = RegistryFileCodec.create(REGISTRY, DIRECT_CODEC);
+        public static final Codec<HolderSet<TestA>> LIST_CODEC = RegistryCodecs.homogeneousList(REGISTRY);
+
+        private final String stringValue;
+        private final int intValue;
+        private final Holder<Biome> biome;
+
+        public TestA(String stringValue, int intValue, Holder<Biome> biome)
+        {
+            this.stringValue = stringValue;
+            this.intValue = intValue;
+            this.biome = biome;
+        }
+
+        public String getStringValue()
+        {
+            return stringValue;
+        }
+
+        public int getIntValue()
+        {
+            return intValue;
+        }
+
+        public Holder<Biome> getBiome()
+        {
+            return biome;
+        }
+    }
+
+    public static class TestB extends ForgeRegistryEntry<TestB>
+    {
+        public static final ResourceKey<Registry<TestB>> REGISTRY = ResourceKey.createRegistryKey(new ResourceLocation(MODID, MODID + "/worldgen/test_b"));
+
+        public static final Codec<TestB> DIRECT_CODEC = RecordCodecBuilder.create(instance -> instance.group(
+                TestA.CODEC.fieldOf("test_a").forGetter(TestB::getTestA),
+                TestA.LIST_CODEC.fieldOf("test_a_list").forGetter(TestB::getTestAList)
+            ).apply(instance, TestB::new));
+
+        private final Holder<TestA> testA;
+        private final HolderSet<TestA> testAList;
+
+        public TestB(Holder<TestA> refB, HolderSet<TestA> refsB)
+        {
+            this.testA = refB;
+            this.testAList = refsB;
+        }
+
+        public Holder<TestA> getTestA()
+        {
+            return testA;
+        }
+
+        public HolderSet<TestA> getTestAList()
+        {
+            return testAList;
+        }
+    }
+    public static class TestC extends ForgeRegistryEntry<TestC>
```

Comment:
Given the consistency in spacing between other members in this file, there should also be a blank line between these two.
```suggestion
    }

    public static class TestC extends ForgeRegistryEntry<TestC>
```